cpp_guidelines/guidelines.md

```diff
@@ -0,0 +1,496 @@
+if(bar)
+{
+    /* something */
+}
+```
+
+### Use spaces, never tabs.
+
+**Reason:** Spaces are the same on all machines, tabs are not.
+
+**Enforced by:**
+  * `.clang-format`: `UseTab: Never`
+
+### The indentation width is set to four columns.
+
+**Reason**: 80 (the maximum line width) is a multiple of four. Two columns do 
+not provide enough whitespace for visual structuring, eight columns (and
+above) fill the 80 columns too fast.
+
+**Enforced by:**
+  * `.clang-format`: `IndentWidth: 4`
+  * `.clang-format`: `TabWidth: 4`
+
+### Align after open brackets.
+
+**Reason:** This aligns multiple parameters nicely that do not fit into a
+single line. This rule applies to `()`, `<>` and `[]`.
+
+**Enforced by:**
+  * `.clang-format`: `AlignAfterOpenBracket: Align`
+  * `.clang-format`: `AllowAllArgumentsOnNextLine: false`
+  * `.clang-format`: `AllowAllConstructorInitializersOnNextLine: false`
+  * `.clang-format`: `AllowAllParametersOfDeclarationOnNextLine: false`
+
+**Example:**
+```c++
+auto some_long_function(argument1,
+                        argument2);
+```
+
+### Align consecutive assignments, bitfields and macros.
+
+**Reason:** It is visually pleasing and there is no reason against it.
+
+**Enforced by:**:
+  * `.clang-format`: `AlignConsecutiveAssignments: true`
+  * `.clang-format`: `AlignConsecutiveBitFields: true`
+  * `.clang-format`: `AlignConsecutiveDeclarations: false`
+  * `.clang-format`: `AlignEscapedNewlines: Right`
+  * `.clang-format`: `BitFieldColonSpacing: Both`
+
+**Example:**
+```c++
+float abcdef = 42.f;
+int xyz      = 3;
+double klmn  = 10.0;
+
+int aaaaaaa : 42;
+int bbbb    : 21;
+int cc      : 10;
+
+#define SHORT                  10
+#define VERY_VERY_LONG         21
+#define THIS_IS_ABSURDELY_LONG 42
+#define evil_func(x)           (x + x)
+#define another_evil_func(x)   (x * x)
+
+#define FOO                             \
+    int aaaaaa;                         \
+    int bbbb;                           \
+    int cc;                             \
+```
+
+### Align operators, colons and commas on the left hand side.
+
+**Reason:** For operators this follows the syntax you would use by writing by
+hand. The commas and colons are kept here for consistency.
+
+**Enforced by:**
+  * `.clang-format`: `AlignOperands: true`
+  * `.clang-format`: `BreakBeforeBinaryOperators: All`
+  * `.clang-format`: `BreakBeforeTernaryOperators: true`
+  * `.clang-format`: `BreakConstructorInitializers: BeforeComma`
+  * `.clang-format`: `BreakInheritanceList: BeforeComma`
+  * `.clang-format`: `OperandAlignmentStyle: Align`
+
+**Example:**
+```c++
+int aaa = bbbbbbbbbb
+        + cccccccccc;
+
+class foo
+    : public first_parent
+    , public second_parent
+{
+    foo()
+        : first_member{some_value}
+        , second_member{another_value}
+};
+```
+
+### Align trailing comments.
+
+**Reason:** This makes the documentation easier to read.
+
+**Enforced by:**
+  * `.clang-format`: `AlignTrailingComments: true`
```

Comment:
I had a bad experience with this option together with `ReflowComments` and short line lengths. Hopefully this has been fixed.